src/theory/quantifiers/sygus/rcons_type_info.cpp

```diff
@@ -0,0 +1,73 @@
+/*********************                                                        */
+/*! \file rcons_type_info.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Abdalrhman Mohamed
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Reconstruct Type Info class implementation
+ **/
+
+#include "theory/quantifiers/sygus/rcons_type_info.h"
+
+#include "theory/datatypes/sygus_datatype_utils.h"
+
+namespace CVC4 {
+namespace theory {
+namespace quantifiers {
+
+void RConsTypeInfo::initialize(TermDbSygus* tds,
+                               SygusStatistics& s,
+                               TypeNode stn,
+                               const std::vector<Node>& builtinVars)
+{
+  NodeManager* nm = NodeManager::currentNM();
+
+  d_enumerator.reset(new SygusEnumerator(tds, nullptr, s, true));
+  d_enumerator->initialize(nm->mkSkolem("sygus_rcons", stn));
+  d_crd.reset(new CandidateRewriteDatabase(true, false, true, false));
+  d_sygusSampler.initialize(stn, builtinVars, 5);
+  d_crd->initialize(builtinVars, &d_sygusSampler);
+}
+
+Node RConsTypeInfo::nextEnum()
+{
+  if (!d_enumerator->increment())
+  {
+    Trace("sygus-rcons") << "no increment" << std::endl;
+    return Node::null();
+  }
+
+  Node sz = d_enumerator->getCurrent();
+
+  if (sz == Node::null())
```

Comment:
Can you reorganize this code so it doesnt check null if the trace is not enabled?  Either inline the null check below, or guard with Trace.isOn("sygus-rcons")